esmvalcore/_main.py

```diff
@@ -105,7 +104,7 @@ def process_recipe(recipe_file, config_user):
     logger.info("If you experience memory problems, try reducing "
                 "'max_parallel_tasks' in your user configuration file.")
 
-    if config_user['compress_netcdf']:
+    if session['compress_netcdf']:
         logger.warning(
             "You have enabled NetCDF compression. Accessing .nc files can be "
             "much slower than expected if your access pattern does not match "
```

Comment:
reminds me of "Negative, Ghost Rider, the pattern is full!" :laughing: (Top Gun)

Reply:
You can read more about HDF5 compression here: https://support.hdfgroup.org/HDF5/doc/Advanced/Chunking/ (and see also here https://docs.xarray.dev/en/stable/user-guide/io.html#chunk-based-compression). What it comes down to is that enabling compression will create compressed chunks of data inside the file. In order to read a single point from a chunk, you need to decompress the entire chunk. So if you first save with chunks that are e.g. one entire lat/lon slice and then try to read a long timeseries from a single lat/lon point this will be very inefficient.

Reply:
I know all that stuff, man, cheers, but a user will not (in 98% of cases) :grin: